src/main/java/io/logz/jmx2logzio/clients/JolokiaClient.java

```diff
@@ -87,73 +116,83 @@ public JolokiaClient(String jolokiaFullURL) {
-
-                Map<String, Object> attrValues = (Map<String, Object>) response.get("value");
-                Map<String, Number> metricToValue = flatten(attrValues);
-                for (String attrMetricName : metricToValue.keySet()) {
-                    try {
-                        metrics.add(new Metric(
-                                attrMetricName,
-                                metricToValue.get(attrMetricName),
-                                metricTime,
-                                dimensions
-                        ));
-                    } catch (IllegalArgumentException e) {
-                        logger.info("Can't sent Metric since it's invalid: "+e.getMessage());
-                    }
-                }
+    private List<Metric> getMetricsForResponse(Map<String, Object> response) {
+        Map<String, Object> request = (Map<String, Object>) response.get(RESPONSE_REQUEST_KEY);
+        String mBeanName = (String) request.get(REQUEST_MBEAN_KEY);
+        int status = (int) response.get(RESPONSE_STATUS_KEY);
+        if (status != HttpURLConnection.HTTP_OK) {
+            String errMsg = "Failed reading mbean '" + mBeanName + "': " + status + " - " + response.get(RESPONSE_ERROR_KEY);
+            logger.warn(errMsg + ". Stacktrace = {}", response.get(RESPONSE_STACKTRACE_KEY));
+            return new ArrayList<>();
+        }
+        Instant metricTime = Instant.ofEpochMilli((int) response.get(RESPONSE_TIMESTAMP_KEY));
+        String[] serviceNameAndArgs = mBeanName.split(":");
+        if (serviceNameAndArgs.length != 2) {
+            logger.debug("metric name {} not valid", mBeanName);
+            return new ArrayList<>();
+        }
+
+        String serviceName = serviceNameAndArgs[SERVICE_NAME_INDEX];
+        String argsString = Metric.DOMAIN_NAME + "=" + serviceName + "," + serviceNameAndArgs[ARGUMENTS_INDEX];
+        List<Dimension> dimensions = Splitter.on(',').splitToList(argsString).stream().map(this::stringArgToDimension).collect(Collectors.toList());
+
+        Map<String, Object> attrValues = (Map<String, Object>) response.get(RESPONSE_VALUE_KEY);
+        Map<String, Number> metricToValue = flatten(attrValues);
+        List<Metric> metrics = new ArrayList<>();
+        for (String attrMetricName : metricToValue.keySet()) {
+            try {
+                metrics.add(new Metric(
+                        attrMetricName,
+                        metricToValue.get(attrMetricName),
+                        metricTime,
+                        dimensions
+                ));
+            } catch (IllegalArgumentException e) {
+                logger.error("Can't send Metric since it's invalid: " + e.getMessage());
             }
+        }
+        return metrics;
+    }
 
-            return metrics;
+    private HttpResponse getHttpResponse(String jolokiaFullURL, String requestBody) {
```

Comment:
It's unclear from the name what the function does. Maybe more like `sendToJolokia`?